### PR TITLE
Deprecate NUM_NODES and NUM_CPUS_PER_NODE in Torque/OpenPBS driver

### LINKED ISSUE #8088: Deprecate NUM_NODES and NUM_CPUS_PER_NODE in OpenPBS driver
The OpenPBS driver, having a legacy from the C Torque driver, has its own keywords picked from the config for specifying CPU resources, namely NUM_NODES and NUM_CPUS_PER_NODE. This is not in line with the LSF driver, Python version or legacy version. 

Having jobs span multiple nodes is probably out of scope for what the Ert application is intended for. If so, NUM_NODES should be deprecated, and then NUM_CPUS_PER_NODE becomes a badly named keyword, and should be shortened to NUM_CPU which already exists.

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #8088 

**Approach**
✂️ 

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [x] **When there are user facing changes**: Updated documentation
- [x] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

**Comment by berland** [comment]
todo: When only NUM_CPU is set in the config, num_cpus_per_node should be set automatcally.

File: test-data/poly_example/poly.ert [hunk 1]
```diff
@@ -1,6 +1,7 @@
 QUEUE_SYSTEM LOCAL
 QUEUE_OPTION LOCAL MAX_RUNNING 50
-
+QUEUE_SYSTEM TORQUE
```
Comment (berland) [inline test-data/poly_example/poly.ert:3]: REMOVE FROM COMMIT, USE git commit -p !

File: src/ert/config/parsing/config_schema_deprecations.py [hunk 2]
```diff
@@ -196,17 +196,14 @@
     ),
     DeprecationInfo(
         keyword="QUEUE_OPTION",
-        message="QUEUE_QUERY_TIMEOUT as QUEUE_OPTION to the TORQUE system will be ignored "
-        "when using the scheduler, and it is not recommended to use this QUEUE_OPTION. "
-        "It has been used in the past to set the time ERT will wait before giving "
-        "up on hanging backend (TORQUE/PBS) when submitting jobs or job status querying.",
+        message="QUEUE_QUERY_TIMEOUT as QUEUE_OPTION is ignored. "
+        "Please remove the line.",
```
Comment (jonathan-eq) [inline src/ert/config/parsing/config_schema_deprecations.py:200]: Nice cleanup! This should probably have been in the commit removing the legacy torque driver :)

File: tests/unit_tests/scheduler/test_openpbs_driver.py [hunk 3]
```diff
@@ -284,24 +284,58 @@ async def started(iens):
...
 
 
+@given(words, st.integers(min_value=1), words)
+@pytest.mark.usefixtures("capturing_qsub")
+async def test_full_resource_string(memory_per_job, num_cpu, cluster_label):
+    driver = OpenPBSDriver(
+        memory_per_job=memory_per_job if memory_per_job else None,
+        cluster_label=cluster_label if cluster_label else None,
+    )
+    await driver.submit(0, "sleep", num_cpu=num_cpu)
```
Comment (jonathan-eq) [inline tests/unit_tests/scheduler/test_openpbs_driver.py:294]: Do we have to submit here, or can we just unit test the `parse_resource_string` function?

File: src/ert/scheduler/openpbs_driver.py [hunk 4]
```diff
@@ -142,16 +142,35 @@ def __init__(
...
             cpu_resources += [f"select={self._num_nodes}"]
         if self._num_cpus_per_node is not None:
-            cpu_resources += [f"ncpus={self._num_cpus_per_node}"]
+            num_nodes = self._num_nodes or 1
+            if num_cpu != self._num_cpus_per_node * num_nodes:
+                raise ValueError(
+                    f"NUM_CPUS_PER_NODE ({self._num_cpus_per_node}) must be equal "
+                    f"to NUM_CPU ({num_cpu}). "
+                    "Please remove NUM_CPUS_PER_NODE from the configuration"
+                )
```
Comment (jonathan-eq) [inline src/ert/scheduler/openpbs_driver.py:171]: Isn't this part already checked for during config validation?

File: tests/unit_tests/scheduler/test_openpbs_driver.py (hunk 3 above)
Comment (berland) [reply tests/unit_tests/scheduler/test_openpbs_driver.py:294]: parse_resources_string depends on the qsub command having run, so yes.

File: src/ert/scheduler/openpbs_driver.py (hunk 4 above)
Comment (berland) [reply src/ert/scheduler/openpbs_driver.py:171]: It is, but in case rogue developers would use the driver independently it should also fail i suppose.

Comment (jonathan-eq) [reply src/ert/scheduler/openpbs_driver.py:171]: Yes, I agree

File: tests/unit_tests/scheduler/test_openpbs_driver.py (hunk 3 above)
Comment (jonathan-eq) [reply tests/unit_tests/scheduler/test_openpbs_driver.py:294]: Ohhh, I thought this was the `build_resource_string` function

File: docs/reference/configuration/queue.rst [hunk 5]
```diff
@@ -292,23 +292,12 @@ The following is a list of all queue-specific configuration options:
...
-  If you wish to increase this number, the program running will usually also
-  have to be told to correspondingly use more processing units (e.g. for
-  ECLIPSE, use the keyword ``PARALLEL``).
-
-  The following should allow 3 × 8 = 24 CPUs for processing realizations::
-
-    QUEUE_SYSTEM TORQUE
-    QUEUE_OPTION TORQUE NUM_NODES 3
-    QUEUE_OPTION TORQUE NUM_CPUS_PER_NODE 8
+  NUM_CPUS_PER_NODE is deprecated, instead only NUM_CPU should be used for
```
[the remark this answers is not in the document]
Comment (berland) [reply docs/reference/configuration/queue.rst]: fixed.

File: src/ert/scheduler/openpbs_driver.py [hunk 6]
```diff
@@ -142,16 +142,35 @@ def __init__(
         self._finished_job_ids: Set[str] = set()
         self._finished_iens: Set[int] = set()
 
-        self._resources = self._resource_strings()
+        if self._num_nodes is not None and self._num_nodes > 1:
+            logger.warning(
```
Comment (xjules) [inline src/ert/scheduler/openpbs_driver.py:146]: Isn't this reported already in here:
```python
 DeprecationInfo(
        keyword="QUEUE_OPTION",
        message="NUM_CPUS_PER_NODE as QUEUE_OPTION to Torque is deprecated and will removed in "
        "the future. Replace by NUM_CPU.",
        check=lambda line: "NUM_CPUS_PER_NODE" in line,
    ),
    DeprecationInfo(
        keyword="QUEUE_OPTION",
        message="NUM_NODES as QUEUE_OPTION to Torque is deprecated and will removed in "
        "the future. Replace by NUM_CPU on a single compute node.",
        check=lambda line: "NUM_NODES" in line,
    ),
```

File: src/ert/scheduler/openpbs_driver.py [hunk 7]
```diff
@@ -142,16 +142,35 @@ def __init__(
...
+    def _build_resource_string(self, num_cpu: int = 1) -> List[str]:
         resource_specifiers: List[str] = []
 
         cpu_resources: List[str] = []
         if self._num_nodes is not None:
             cpu_resources += [f"select={self._num_nodes}"]
         if self._num_cpus_per_node is not None:
-            cpu_resources += [f"ncpus={self._num_cpus_per_node}"]
+            num_nodes = self._num_nodes or 1
+            if num_cpu != self._num_cpus_per_node * num_nodes:
```
Comment (xjules) [inline src/ert/scheduler/openpbs_driver.py:166]: This is checked by the config validation already

Comment (berland) [reply src/ert/scheduler/openpbs_driver.py:166]: Yes, @jonathan-eq  also pointed that out but we agreed to keep it here too.

File: src/ert/scheduler/openpbs_driver.py (hunk 6 above)
Comment (berland) [reply src/ert/scheduler/openpbs_driver.py:146]: Correct. The config system will flash it in the GUI, while this will leave a trace in the jobqueue log.

**Review by xjules — APPROVED** [review]
Good job @berland ! :rocket:

**Pull request merged**